experimental/dds/tree2/src/shared-tree-core/editManager.ts

```diff
@@ -257,8 +257,20 @@ export class EditManager<
 		const searchResult = this.sequenceMap.getPairOrNextHigher(trunkTailSearchKey);
 		if (searchResult !== undefined) {
 			const [_, newTrunkTail] = searchResult;
-			// Don't do any work if the commit found by the search is already the tail of the trunk
+			// Only work if the commit found by the search is farther than the current tail of the trunk
 			if (newTrunkTail.parent !== this.trunkBase) {
+				// The minimum sequence number informs us that all peer branches are at least caught up to the tail commit,
+				// so rebase them accordingly. This is necessary to prevent peer branches from referencing any evicted commits.
+				for (const [sessionId, branch] of this.peerLocalBranches) {
+					const [rebasedBranch] = rebaseBranch(
```

Comment:
Note that if the branch is already rebased at or past the tail (which is common), then `rebaseBranch` will early-out very quickly. So it should not be expensive to "rebase every peer branch" every time we evict.